lib/pdf/core/renderer.rb

```diff
@@ -216,10 +217,12 @@ def render_xref(output)
       # Write out the PDF Trailer, as per spec 3.4.4
       #
       def render_trailer(output)
+        trailer_id = PDF::Core::ByteString.new(SecureRandom.random_bytes(16))
```

Comment:
Let's make is deterministic. We're trying to reduce random changes to the generated documents.

Reply:
I'm looking at getting both PDF/A-1b support in, and updating doc encryption to modern levels.  In the process, I found that there's a potential issue here that will likely force a design decision for Prawn.

Specifically, the first trailer[:ID] value is used in calculating the encryption key for the PDF document.  This is missing from the current Prawn implementation (which always assumes an empty :iD value), but is clearly laid out in the spec.  So this ID value needs to be present when the encryption dictionary is built (currently when `encrypt_document` is called) and invariant thereafter.

My current favored approach is:

1. Calling `encrypt_document` should initialize the trailer[:iD] value in the exact same way as the PDF/A-1b code does, with the caveat that they may not wind up with the same value if the state of the document is altered between the two calls.  Also, need to confirm that the current method for PDF/A-1b excludes an encryption dictionary from the ID calculation.
2. The PDF/A-1b code only sets trailer[:ID] if it isn't already set.  Otherwise it uses the already present value

I think this is simple, involves the minimum change for Prawn and PDF-Core, and is consistent with the spec intentions.

Other options might include moving all encryption_dictionary calculation to the render phase or requiring that `encrypt_document` be the last call to a document before rendering.  On the face of it both of those seem less desirable so I haven't yet thought through the implications.

@pointlessone Thoughts?